AntennaTracker/mode_auto.cpp

```diff
@@ -4,9 +4,48 @@
 
 void ModeAuto::update()
 {
+    State old_state = _state;
+
     if (tracker.vehicle.location_valid) {
         update_auto();
-    } else if (tracker.target_set || (tracker.g.auto_opts.get() & (1 << 0)) != 0) {
+        _state = target_range_acceptable() ? State::TRACKING : State::SUPPRESSED;
+    } else if (tracker.target_set || scan_for_any_target()) {
         update_scan();
+        _state = State::SCANNING;
+    } else {
+        _state = State::IDLE;
+    }
+
+    // Announce the state change if necessary
+    if (old_state != _state) {
+        switch (_state) {
+            case State::TRACKING:
+                gcs().send_text(MAV_SEVERITY_INFO, "Tracker locked on the vehicle %d", (short)tracker.g.sysid_target);
+                break;
+
+            case State::SCANNING:
+                if (scan_for_any_target()) {
+                    gcs().send_text(MAV_SEVERITY_INFO, "Tracker scanning for a vehicle");
+                } else {
+                    gcs().send_text(MAV_SEVERITY_INFO, "Tracker scanning for the vehicle %d", (short)tracker.g.sysid_target);
+                }
+                break;
+
+            case State::SUPPRESSED:
+                gcs().send_text(MAV_SEVERITY_INFO, "Tracker suppressed while the target is within %d m", (short)tracker.g.distance_min);
+                break;
+
+            case State::IDLE:
+                gcs().send_text(MAV_SEVERITY_INFO, "Tracker idle");
+                break;
+
+            default:
+                break;
+        }
     }
 }
+
+bool ModeAuto::scan_for_any_target() const
+{
+    return (tracker.g.auto_opts.get() & (1 << 0)) != 0;
+}
```

Comment:
If your feeling keen we typically do this sort of thing with a `option_is_set` helper and a enum for the options. There are lots of examples.